src/sagemaker/experiments/_metrics.py

```diff
@@ -0,0 +1,413 @@
+        """Get file path to store metrics"""
+        pid_filename = "{}.json".format(str(os.getpid()))
+        metrics_file_path = self._metrics_file_path or os.path.join(METRICS_DIR, pid_filename)
+        logger.debug("metrics_file_path = %s", metrics_file_path)
+        return metrics_file_path
+
+
+class SageMakerMetricsWriterException(Exception):
+    """SageMakerMetricsWriterException"""
+
+    def __init__(self, message, errors=None):
+        """Construct a `SageMakerMetricsWriterException` instance"""
+        super().__init__(message)
+        if errors:
+            self.errors = errors
+
+
+class _RawMetricData(object):
+    """A Raw Metric Data Object"""
+
+    MetricName = None
+    Value = None
+    Timestamp = None
+    Step = None
+
+    def __init__(self, metric_name, value, timestamp=None, step=None):
+        """Construct a `_RawMetricData` instance.
+
+        Args:
+            metric_name (str): The name of the metric.
+            value (float): The value of the metric.
+            timestamp (datetime.datetime or float or str): Timestamp of the metric.
+                If not specified, the current UTC time will be used.
+            step (int):  Iteration number of the metric (default: None).
+        """
+        if timestamp is None:
+            timestamp = time.time()
+        elif isinstance(timestamp, datetime.datetime):
+            # If the input is a datetime then convert it to UTC time.
+            # Assume a naive datetime is in local timezone
+            if not timestamp.tzinfo:
+                timestamp = timestamp.replace(tzinfo=dateutil.tz.tzlocal())
+            timestamp = (timestamp - timestamp.utcoffset()).replace(tzinfo=datetime.timezone.utc)
+            timestamp = timestamp.timestamp()
+        else:
+            timestamp = float(timestamp)
+
+        if timestamp < (time.time() - METRIC_TS_LOWER_BOUND_TO_NOW) or timestamp > (
+            time.time() + METRIC_TS_UPPER_BOUND_FROM_NOW
+        ):
+            raise ValueError(
+                "Supplied timestamp %f is invalid."
+                " Timestamps must be between two weeks before and two hours from now." % timestamp
+            )
+        value = float(value)
+
+        self.MetricName = metric_name
+        self.Value = float(value)
+        self.Timestamp = timestamp
+        if step is not None:
+            if not isinstance(step, int):
+                raise ValueError("step must be int.")
+            self.Step = step
+
+    def to_record(self):
+        """Convert the `_RawMetricData` object to dict"""
+        return self.__dict__
+
+    def to_raw_metric_data(self):
+        """Converts the metric data to a BatchPutMetrics RawMetricData item"""
+        # Convert timestamp from float to timestamp str.
+        # Otherwise will get ParamValidationError
+        raw_metric_data = {
+            "MetricName": self.MetricName,
+            "Value": self.Value,
+            "Timestamp": str(int(self.Timestamp)),
```

Comment:
looks like the str is not needed. likely you added this because your environment had a different model?

```
Invalid type for parameter MetricData[0].Timestamp, value: 1670999607, type: <class 'str'>, valid types: <class 'int'>
```

Reply:
nm this was an issue with my env. fwiw eventually we want to remove the `int` since that cuts off the decimal which would indicate milliseconds